packages/cozy-client/src/hooks/useClientErrors.spec.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import useClientErrors from './useClientErrors'
+import { renderHook, act } from '@testing-library/react-hooks'
+import CozyProvider from '../Provider'
+import { shallow } from 'enzyme'
+import { FetchError } from 'cozy-stack-client'
+
+function createCozyClient() {
+  return {
+    on: jest.fn(),
+    removeListener: jest.fn()
+  }
+}
+
+function createWrapper(client = createCozyClient()) {
+  function Wrapper({ children }) {
+    return <CozyProvider client={client}>{children}</CozyProvider>
+  }
+  return Wrapper
+}
+
+function renderWrappedHook(params, client) {
+  const wrapper = createWrapper(client)
+  return renderHook(() => useClientErrors(), { wrapper })
+}
+
+function wrappedShallow(tree, client) {
+  return shallow(tree, { wrappingComponent: createWrapper(client) })
+}
+
+describe('useClientErrors', () => {
+  it('registers an `error` handler in client', done => {
+    const client = createCozyClient()
+    client.on.mockImplementation(name => name === 'error' && done())
+    renderWrappedHook(undefined, client)
+  })
+
+  describe('renderErrors', () => {
+    it('returns a function', () => {
+      const { result } = renderWrappedHook()
+      const { ClientErrors } = result.current
+      expect(ClientErrors).toBeInstanceOf(Function)
+    })
+
+    it('displays nothing by default', () => {
+      const { result } = renderWrappedHook()
+      const { ClientErrors } = result.current
+      const node = wrappedShallow(<ClientErrors />)
+      expect(node).toHaveLength(0)
+    })
+
+    describe('for quota errors', () => {
+      it('displays a modal', async () => {
+        const client = createCozyClient()
+        const { result, waitForNextUpdate } = renderWrappedHook(
+          undefined,
+          client
+        )
+        const nextUpdate = waitForNextUpdate()
+
+        const response = new Response(null, {
+          status: 413,
+          statusText: 'Quota exceeded'
+        })
+        const error = new FetchError(
+          response,
+          `${response.status} ${response.statusText}`
+        )
+        act(() => {
+          const handler = client.on.mock.calls[0][1]
+          handler(error)
+        })
+
+        await nextUpdate
+        const { ClientErrors } = result.current
+        const node = wrappedShallow(<ClientErrors />, client)
+        expect(node).toHaveLength(1)
```

Comment:
Could we test that QuotaAlert is displayed ?

Reply:
I didn't managed to. The component from UI doesn't keep its displayname. I could `mount` and check if its a modal, but not the actual name of the component

Reply:
expect(mount(<ClientErrors />).find(QuotaAlert).length).toBe(1)

should work no ?

Reply:
nope, QuotaAlert is seen as "Component". Let's see that tomorow morning

Reply:
I guess since QuotaAlert is declared with `withClient`, the JSX transpiler cannot infer its name, I think by postponing withClient to the `export default` site, QuotaAlert will have the right name.